storage/aptosdb/src/lib.rs

```diff
@@ -1766,12 +1796,36 @@ impl DbWriter for AptosDB {
                     num_transactions_in_db,
                 );
 
-                // Persist ledgerDB data first.
+                // Commit multiple batches for different DBs in parallel, then write the overall
+                // progress.
                 {
                     let _timer = OTHER_TIMERS_SECONDS
                         .with_label_values(&["save_transactions_commit"])
                         .start_timer();
-                    self.commit(batch)?;
+                    state_kv_batch.put::<DbMetadataSchema>(
+                        &DbMetadataKey::StateKVCommitProgress,
+                        &DbMetadataValue::Version(last_version),
+                    )?;
+
+                    ledger_batch.put::<DbMetadataSchema>(
+                        &DbMetadataKey::LedgerCommitProgress,
+                        &DbMetadataValue::Version(last_version),
+                    )?;
+
+                    thread::scope(|s| {
+                        let t0 = s.spawn(|| self.state_kv_db.write_schemas(state_kv_batch));
+                        let t1 = s.spawn(|| self.ledger_db.write_schemas(ledger_batch));
+                        t0.join().unwrap().unwrap();
+                        t1.join().unwrap().unwrap();
+                    });
+
+                    let ledger_batch = SchemaBatch::new();
+                    self.save_ledger_info(new_root_hash, ledger_info_with_sigs, &ledger_batch)?;
+                    ledger_batch.put::<DbMetadataSchema>(
+                        &DbMetadataKey::OverallCommitProgress,
```

Comment:
curious about the motivation for introducing OverallCommitProgress?

Reply:
It's easier to understand and maintain.
Also, one reason I don't use the LedgerInfo version as the progress here because we overwrite the previous LedgerInfo in the same epoch in every commit. If we use the db truncation tool to truncate the db, the ledger info version will not be the same as the target version of the truncation. In this case I'd rely on this OverallCommitProgress.

Reply:
https://github.com/aptos-labs/aptos-core/pull/6482